src/Utilities/FlowAnalysis/Framework/DataFlow/AnalysisEntityMapAbstractDomain.cs

```diff
@@ -111,7 +111,7 @@ TValue GetMergedValueForEntityPresentInOneMap(AnalysisEntity key, TValue value)
 
                 if (!resultMap.ContainsKey(key1))
                 {
-                    resultMap[key1] = ValueDomain.UnknownOrMayBeValue;
+                    resultMap[key1] = GetMergedValueForEntityPresentInOneMap(key1, value1);
```

Comment:
Good idea or bad idea?

resultMap was ending up with TaintedDataAbstractValueKind.Unknown entries, and Unknown isn't at the top of the lattice.  So later when comparing map1 to resultMap, I'd get a non-monotonic merge assert comparing TaintedDataAbstractValueKind.Tainted to Unknown.

Reply:
Seems like the correct fix to me. Thanks!

---
In reply to: [233217454](https://github.com/dotnet/roslyn-analyzers/pull/1882#discussion_r233217454) [](ancestors = 233217454)